src/TritonRoute/src/pa/FlexPA_prep.cpp

```diff
@@ -816,10 +816,11 @@ void FlexPA::prepPoint_pin_checkPoint_via(
   auto upper_type = ap->getType(false);
   if (layerNum >= VIAINPIN_BOTTOMLAYERNUM && layerNum <= VIAINPIN_TOPLAYERNUM) {
     viainpin = true;
-  } else if ((lower_type == frAccessPointEnum::EncOpt
+  } else if (VIAINPIN_BOTTOMLAYERNUM == VIAINPIN_DEFAULT_BOTTOMLAYERNUM && 
```

Comment:
Why do we need this check?

Reply:
Because if you set the Via-in-pin to anything  other than its default value (which is infinity), failing the check of line 817 implies that viainpin should be false.

Reply:
If your type is frAccessPointEnum::EncOpt that is equivalent to viainpin?  That seems to be what the code used to say.

Reply:
Almost. The old code considered that, if the type is frAccessPointEnum::EncOpt in one lyer and not frAccessPointEnum::NearbyGrid in the other one, viainpin was true. But this is wrong, because if you set the VIAINPIN_BOTTOMLAYERNUM  flag and it fails the first if, then viainpin should be false. 
viainpin can be forced either by VIAINPIN_BOTTOMLAYERNUM and VIAINPIN_TOPLAYERNUM flags or by frAccessPointEnum options of the down if. The problem is that if the first if fails it can be either because VIAINPIN_BOTTOMLAYERNUM is infinity, i.e., it was not set, or because it was set and layerNum is out of range. In this last case, viainpin should be necessarily false, not matter frAccessPointEnum value.

Reply:
I was considering that if the via-in-pin flags are set then viainpin should be true only in the layer is whithin the layer range from the via-in-pin flags. But I think now that this assumption is not true. In this case, this test is unnecessary. I will remove it.